### PR TITLE
User settings for email, isVerified; Change email; Fix app error handling

### PR DESCRIPTION
**Pull request by gabriel**
Plumbs through email and isVerified for settings UI.

I also removed the `VisibleTab` type, and rename `Tabs` type to `Tab`. The visibility of a tab should be a prop not a type.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers for js
@maxtaco for go

**Comment by gabriel** [comment]
@keybase/react-hackers  Created [issue](https://keybase.atlassian.net/browse/DESKTOP-2084) to have the UI wait for bootstrap/login state before showing main UI, since settings and other UI assumes this state is set.

**Comment by gabriel** [comment]
@keybase/react-hackers  Including change email plumbing in this PR since it depends on this earlier PR

File: shared/actions/plan-billing.js [hunk 1]
```diff
@@ -173,17 +172,6 @@ function * fetchBillingAndQuotaSaga (): SagaGenerator<any, any> {
...
-  const loggedInSelector = ({config: {loggedIn}}: TypedState) => loggedIn
-
-  let bootstrapStatus: BootStatus = ((yield select(overallBootstrappedSelector)): any)
-  let loggedIn: boolean = ((yield select(loggedInSelector)): any)
-
-  while (bootstrapStatus !== 'bootStatusBootstrapped' || !loggedIn) {
-    yield call(delay, 500)
-    bootstrapStatus = ((yield select(overallBootstrappedSelector)): any)
-    loggedIn = ((yield select(loggedInSelector)): any)
-  }
```
Comment (MarcoPolo) [inline shared/actions/plan-billing.js]: Shouldn't you check that you are logged in & bootstrapped, and if you aren't you wait for the logged in and bootstrap action before making the call ?

File: shared/actions/settings.js [hunk 2]
```diff
@@ -77,6 +89,24 @@ function * _onUpdatePGPSettings (): SagaGenerator<any, any> {
   }
 }
 
+function * _onSubmitNewEmail (): SagaGenerator<any, any> {
+  try {
+    const selector = (state: TypedState) => state.settings.email
```
Comment (MarcoPolo) [inline shared/actions/settings.js]: have the selector return newEmail instead. Since it's typed, you are sure you'll get newEmail

File: shared/actions/settings.js [hunk 3]
```diff
@@ -77,6 +89,24 @@ function * _onUpdatePGPSettings (): SagaGenerator<any, any> {
...
+    const {newEmail} = ((yield select(selector)): any)
+    yield call(accountEmailChangeRpcPromise, {
+      param: {
+        newEmail,
+      },
+    })
+    // Reload settings
+    const userSettings = yield call(userLoadMySettingsRpcPromise)
```
Comment (MarcoPolo) [inline shared/actions/settings.js]: You should just `put(loadSettings())`.

File: shared/settings/email/container.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import React, {Component} from 'react'
+import UpdateEmail from './index'
+import {connect} from 'react-redux'
+import {navigateUp} from '../../actions/router'
+import {onSubmitNewEmail, onChangeNewEmail} from '../../actions/settings'
+
+import type {Props} from './index'
+import type {TypedState} from '../../constants/reducer'
+
+class UserEmailContainer extends Component<void, Props, void> {
```
Comment (MarcoPolo) [inline shared/settings/email/container.js]: 🐑 You can use the Routable HOC

It would look like:

```
const UserEmailContainer = Routable(() => ({componentAtTop: {title: 'Change Email'}}), UpdateEmail)
```
In util/routable

File: shared/settings/email/container.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export default connect(
+  (state: TypedState, ownProps: {}) => {
+    const {emails, errorMessage, newEmail} = state.settings.email
+    if (emails.length > 0) {
+      const email = emails[0].email
+      return {
+        email,
+        isVerified: emails[0].isVerified,
```
Comment (MarcoPolo) [inline shared/settings/email/container.js]: 🐑 email.isVerified

File: shared/settings/email/container.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        edited: newEmail && newEmail !== email,
+        errorMessage,
+      }
+    }
+    return {}
```
Comment (MarcoPolo) [inline shared/settings/email/container.js]: I'm sorry this connect isn't type checking properly. I need to fix that.
But you need to return the right keys with null values.
Instead of emptyObj, should have `{errorMessage: null, edited: false...}` Needs to match the type def.

File: shared/settings/landing/container.js [hunk 7]
```diff
@@ -1,53 +1,64 @@
...
+    }
+
     const {planBilling: {availablePlans, usage, plan, paymentInfo}} = state
-    if (!availablePlans || !usage || !plan) {
-      return {
-        bootstrapDone: false,
+    let planProps
+    if (plan && usage) {
+      const freeSpaceGB = plan.gigabytes - usage.gigabytes
+      const freeSpacePercentage = freeSpaceGB / plan.gigabytes
```
Comment (MarcoPolo) [inline shared/settings/landing/container.js]: Might want to pretty these numbers a bit so you don't end up with .6666666 ...

File: shared/settings/landing/container.js [hunk 8]
```diff
@@ -1,53 +1,64 @@
...
+        paymentInfo,
+        onChangePaymentInfo: () => console.log('todo'),
       }
     }
 
-    const freeSpaceGB = plan.gigabytes - usage.gigabytes
-    const freeSpacePercentage = freeSpaceGB / plan.gigabytes
+    // When enabling planProps, we should check both for bootstrapDone:
+    // let bootstrapDone = accountProps && planProps
+    let bootstrapDone = !!accountProps
```
Comment (MarcoPolo) [inline shared/settings/landing/container.js]: You're only checking accountProps, but the comment says you are checking planProps too?

File: shared/constants/tabs.js [hunk 9]
```diff
@@ -28,18 +28,15 @@ const prettyNames = {
   [profileTab]: 'Profile',
 }
 
-export type VisibleTab = ProfileTab
```
Comment (MarcoPolo) [inline shared/constants/tabs.js]: Why this change? It's nice to have a type for visible tabs.

File: shared/actions/settings.js (hunk 2 above)
Comment (gabriel) [inline shared/actions/settings.js]:
```
const newEmail = yield select(state => state.settings.email.newEmail)
```
?

File: shared/settings/email/container.js (hunk 5 above)
Comment (gabriel) [inline shared/settings/email/container.js]: Will do `const {email, isVerified} = emails[0]`

File: shared/actions/settings.js (hunk 3 above)
Comment (gabriel) [inline shared/actions/settings.js]: Fixed.

File: shared/settings/email/container.js (hunk 4 above)
Comment (gabriel) [inline shared/settings/email/container.js]: Fixed.

File: shared/settings/email/container.js (hunk 6 above)
Comment (gabriel) [inline shared/settings/email/container.js]: Fixed.

File: shared/settings/landing/container.js (hunk 7 above)
Comment (gabriel) [inline shared/settings/landing/container.js]: I was moving this code. We can deal with that when we bring it back?

File: shared/settings/landing/container.js (hunk 8 above)
Comment (gabriel) [inline shared/settings/landing/container.js]: Currently it only waits for account props cause plans aren't being loaded anymore (that was temporarily removed from view).

**Comment by gabriel** [comment]
Moved https://github.com/keybase/client/pull/4712 into here since @MarcoPolo was same reviewer.

**Comment by gabriel** [comment]
@MarcoPolo Read for final review

File: shared/settings/landing/container.js (hunk 7 above)
Comment (MarcoPolo) [inline shared/settings/landing/container.js]: 👍

File: shared/settings/landing/container.js (hunk 8 above)
Comment (MarcoPolo) [inline shared/settings/landing/container.js]: Maybe leave a comment?

File: shared/actions/settings.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/actions/settings.js]: No, this is worse since the selector fn `state => state.settings.email.newEmail` won't have any type safety
